pkg/kubectl/resource_printer.go

```diff
@@ -84,15 +86,15 @@ func GetPrinter(format, formatArgument string) (ResourcePrinter, bool, error) {
 // ResourcePrinter is an interface that knows how to print runtime objects.
 type ResourcePrinter interface {
 	// Print receives a runtime object, formats it and prints it to a writer.
-	PrintObj(runtime.Object, io.Writer) error
+	PrintObj(runtime.Object, *watch.EventType, io.Writer) error
 }
 
 // ResourcePrinterFunc is a function that can print objects
-type ResourcePrinterFunc func(runtime.Object, io.Writer) error
+type ResourcePrinterFunc func(runtime.Object, *watch.EventType, io.Writer) error
```

Comment:
This PR is changing a ton of interfaces that is reused by OpenShift.  My recommendations are targeted towards keeping this change (which is invasive today) focused around solving the core problem that the resource printers are not properly using composition to deal with orthogonal concerns (the event printer should not know about how to print the wrapped event object, only that it wants to add an event type column to whatever columns the wrapped object has).  So no, it is not strictly required to fix this problem, but any signature change to these methods causes a large amount of work to be generated for folks that reuse this code.  I would prefer to try to get to a point where we stop having to change these signatures every time we come up with a new use case for printing.